generator_expression.py

```diff
@@ -0,0 +1,15 @@
+numbers = [1,2,3,4,5]
+#a generator expression(note:This is not a tuple comprehension)
+lazy_squares = (x*x for x in numbers)
+print(type(lazy_squares))
+for i in numbers:
```

Comment:
explicit callimg of next is one example:

you can also show the implicit calling of next:
eg.
<pre>
for i in lazy_squares:
     print i
</pre>
here "next" is implicitly called by the for statement and the reutrn value of next is stored in "i"